sys-utils/nsenter.c

```diff
@@ -187,6 +189,56 @@ static int is_usable_namespace(pid_t target, const struct namespace_file *nsfile
 	return true; /* All pass */
 }
 
+static char** get_process_environment(void) {
+	char *buffer = NULL, *p = NULL;
+	size_t size = 0;
+	ssize_t bytes_read;
+	char** envp;
+	int num;
+
+	do {
+		buffer = (char*)realloc(buffer, size + 4096);
+		if (buffer == NULL) {
+			err(EXIT_FAILURE, _("allocate memory failed"));
+			close(environ_fd);
+			return NULL;
+		}
+		bytes_read = read(environ_fd, buffer + size, 4096);
+		if (bytes_read == -1) {
+			err(EXIT_FAILURE, _("read file failed"));
```

Comment:
err() will exit the process, no need for cleanup/return

Reply:
Thank you for pointing it out.
Updated